src/NewTools-Inspector-Tests/StInspectorTest.class.st

```diff
@@ -23,6 +23,84 @@ StInspectorTest >> tearDown [
 	super tearDown
 ]
 
+{ #category : #tests }
+StInspectorTest >> testBuildPresentationWithDoesNotUndertandShouldReturnCodePresenter [
+
+
+	| collector pragma context |
+	collector := StInspectionCollector on: StInspectorMockObject new.
+	pragma := (StInspectorMockObject >> #methodWithDnuingPresentation) pragmas first.
+
+	context := collector contextFromPragma: pragma.
+	
+	"Smoke test -> creating an inspection presenter should not fail. Instead it should create an error presenter"
+	context newInspectionPresenter
```

Comment:
Missing assertion on the returned presenter type (same remark for next 2 tests)? 
```Smalltalk
self assert: context newInspectionPresenter isErrorPresenter
```

Reply:
This test does not return an error presenter but a normal CodePresenter instance. So testing it is much more difficult. What I could test is that it is a code presenter that contains code that when evaluated will reproduce the exception. That is doable, worth it?

Reply:
Checking the code looks a bit complex for an assertion. 
Maybe the test name confused me. If it was named testBuildPresentationWithDoesNotUndertandShouldNotRaiseError, it would be ok for me.